### DISCUSSION TITLE
Seeing from gradCAM that the model cannot focus on identifying in non-background (non-zero) regions

### DISCUSSION THREAD

**Discussion by Minxiangliu** [Project-MONAI/MONAI#4086] [Q&A] [answered]
I used the [nvidia pre-training model](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/med/models/clara_pt_chest_xray_classification) to perform transfer learning on the cropped images of the lungs. These images have been processed so that the background is 0. 
The random transforms during the training process is as follows:
```
RandScaleIntensityd(keys=['image'], factors=(0.1, 0.3), prob=0.5),
RandFlipd(keys=['image'], spatial_axis=[0, 1], prob=0.5),
RandRotated(keys=['image'], range_x=[-0.5, 0.5], prob=0.5),
RandCoarseDropoutd(keys=['image'], spatial_size=(10,10), fill_value=0, holes=100, prob=0.5)
```

After training I get decent accuracy, but from gradCAM I see that the eigenvalue of the background is 1, what am I doing wrong?
I expect the model to pay attention to only non-zero lung regions, is this possible?
![image](https://user-images.githubusercontent.com/30144428/162190787-79a8de96-13f8-4bea-b602-59d2919280c6.png)

[Image:](https://drive.google.com/file/d/17nEOjYZOJsaB8hJF1SrH2hJJsAqxBHMs/view?usp=sharing)
![image](https://user-images.githubusercontent.com/30144428/162190176-089b121a-0a01-40e4-86fe-eabe4ada19d4.png)

[chcekpoint](https://drive.google.com/file/d/1kIRvpOG3aXnSnirLp5IIlPo9Aa3HoXW6/view?usp=sharing)

Code:
```
import matplotlib.pyplot as plt
import torch
from monai.networks.nets import DenseNet121
from monai.visualize import GradCAM, blend_images
from monai.transforms import (
    Compose,
    LoadImaged,
    AddChanneld,
    Rotate90d,
    ScaleIntensityd,
    EnsureTyped,
    Flipd
    )

checkpoint = torch.load('epoch=0090-val_acc=0.97.ckpt', map_location='cuda')
state_dict = checkpoint['state_dict']
for k in list(state_dict.keys()):
    if k.startswith('model.'):
        if k.startswith('model'):
            state_dict[k[len("model."):]] = state_dict[k]
    del state_dict[k]

model = DenseNet121(spatial_dims=2, in_channels=1, out_channels=1).cuda()
log = model.load_state_dict(state_dict, strict=True)

cam = GradCAM(nn_module=model, target_layers='class_layers.relu')

trans = Compose([LoadImaged(keys=['image']),
                 AddChanneld(keys=['image']),
                 Rotate90d(keys=['image'], k=3),
                 Flipd(keys=['image'],spatial_axis=1),
                 ScaleIntensityd(keys=['image']),
                 EnsureTyped(keys=['image'])])

for data in trans([{'image':'tb0997.png'}]):
    cam_result = cam(x=data['image'].unsqueeze(0).cuda())

img = data['image'].cpu()
heatmap = cam_result.squeeze(0).cpu()

blend_img = blend_images(image=img, label=heatmap, alpha=0.5, cmap="hsv", rescale_arrays=False)
blend_img = blend_img.moveaxis(0,-1)

fig, axes = plt.subplots(1, 3, figsize=(10, 10), facecolor='white')

im_0 = axes[0].imshow(img.squeeze(), cmap='gray')
axes[0].axis('off')
fig.colorbar(im_0, ax=axes[0], shrink=0.25)

im_1 = axes[1].imshow(heatmap.squeeze(), cmap='jet')
axes[1].axis('off')
fig.colorbar(im_1, ax=axes[1], shrink=0.25)

im_2 = axes[2].imshow(blend_img)
axes[2].axis('off')
fig.colorbar(im_2, ax=axes[2], shrink=0.25)

plt.savefig('out.png')
    
```

**Comment by rijobro** [comment, marked as answer]
I think the GradCAM is saying that whatever decision it is making, the blue area was the most important in the decision-making process. So the bottom of the left-side lung. What seems to be the problem here, and what classification task are you doing?

You could try running with occlusion sensitivity to compare results.

**Reply by Minxiangliu** [reply]
Thank you for your reply, I am doing `binary classification` of tuberculosis, and what does occlusion sensitivity mean?

**Reply by rijobro** [reply]
Both GradCAM and occlusion sensitivity are saliency methods. They help you understand why your network made certain decisions. So in your case, your network made a decision -- either tuberculosis or not -- and then the blue region of the GradCAM is telling you which area of the image was important in reaching that decision. Occlusion sensitivity shows the same sort of information, but computed in a different way, so can show complementary information. See [this explanation here](https://uk.mathworks.com/help/deeplearning/ug/understand-network-predictions-using-occlusion.html) for occlusion sensitivity.

Also see [our interpretability tutorials](https://github.com/Project-MONAI/tutorials/tree/master/modules/interpretability) for examples of both GradCAM and occlusion sensitivity. I'd recommend starting with dogs and cats.
